js/web/lib/wasm/jsep/webgpu/ops/where.ts

```diff
@@ -74,8 +74,8 @@ const createWhereOpProgramInfo = (inputs: readonly TensorView[]): ProgramInfo =>
   const outputDataType = inputs[1].dataType;
 
   const isBroadcast = !(ShapeUtil.areEqual(dimsA, dimsB) && ShapeUtil.areEqual(dimsB, dimsC));
-  let outputShape = dimsA;
-  let outputSize = ShapeUtil.size(dimsA);
+  let outputShape = dimsC;
+  let outputSize = ShapeUtil.size(dimsC);
```

Comment:
Can you explain why `dimsA` -> `dimsC` can fix you issue? It seems no difference for me. If `isBroadcast` is false. dimsA will be equal to dimsC. Otherwise, outputShape will be updated in L87 and replaced by `calculatedShape`.

Reply:
It seems that `vecSize` is not updated when `isBroadcast` is true. Would you please try moving `const vecSize = Math.ceil(outputSize / 4);` to L90 after `isBroadcast` checking to see whether it can fix your issue?

Reply:
where(A, B, C)
The issue is that if A is say [0], outputSize = ShapeUtil.size(dimsA) would be 1,
which becomes vec_size and in main we'd not process the full tensor:
 if (global_idx >= uniforms.vec_size) { return; }

Reply:
> const vecSize = Math.ceil(outputSize / 4);
sure, that should work too

Reply:
Thanks for the explanation. `vecSize` is not updated when bcast is true is the root cause. So I prefer the fixing is to move `const vecSize = Math.ceil(outputSize / 4);` to the right location. Otherwise, we may still have bug if the vecSize is decided by `dimsB` and bcast is true.

Reply:
done